src/Compilers/Core/Portable/Text/ChangedText.cs

```diff
@@ -299,7 +299,7 @@ private static ImmutableArray<TextChangeRange> Merge(ImmutableArray<TextChangeRa
                         AddRange(list, new TextChangeRange(oldChange.Span, oldChangeLeadingInsertion));
                         oldDelta = oldDelta - oldChange.Span.Length + oldChangeLeadingInsertion;
                         oldChange = new TextChangeRange(new TextSpan(oldChange.Span.Start, 0), oldChange.NewLength - oldChangeLeadingInsertion);
-                        newChange = new TextChangeRange(new TextSpan(oldChange.Span.Start + oldDelta, newChange.Span.Length), newChange.NewLength);
+                        //newChange = new TextChangeRange(new TextSpan(oldChange.Span.Start + oldDelta, newChange.Span.Length), newChange.NewLength);
```

Comment:
I think this is right. I can't convince myself why I thought that newChange needed to be modified here.

Reply:
Or why that was the start position..  It may have been left over from a previous pass where the new change was being added inline here and needed to be adjusted, but now that gets handled during the retry loop, so the delta may be getting applied doubly.